src/spring/azext_spring/_validators_enterprise.py

```diff
@@ -215,6 +220,37 @@ def _validate_sso(namespace):
         namespace.scope = namespace.scope.split(",") if namespace.scope else []
 
 
+def _validate_gateway_apm_types(namespace):
+    if namespace.apm_types is None:
+        return
+    allowedApmTypes = [v20221101_preview_AppPlatformEnums.GatewayApmType.APPLICATION_INSIGHTS,
+                       v20221101_preview_AppPlatformEnums.GatewayApmType.APP_DYNAMICS,
+                       v20221101_preview_AppPlatformEnums.GatewayApmType.DYNATRACE,
+                       v20221101_preview_AppPlatformEnums.GatewayApmType.NEW_RELIC,
+                       v20221101_preview_AppPlatformEnums.GatewayApmType.ELASTIC_APM]
+    for type in namespace.apm_types:
+        if (type not in allowedApmTypes):
+            raise InvalidArgumentValueError("Allowed APM types are " + ','.join(allowedApmTypes))
```

Comment:
How about using the `v20221101_preview_AppPlatformEnums.GatewayApmType` enumeration from Python SDK directly?